src/metadata/trackmetadatataglib.cpp

```diff
@@ -0,0 +1,688 @@
+                        toQString(replaygainFrame->fieldList()[1]));
+            }
+            // Prefer track gain over album gain.
+            if (desc == "replaygain_track_gain") {
+                pTrackMetadata->setReplayGainDbString(
+                        toQString(replaygainFrame->fieldList()[1]));
+            }
+        }
+    }
+
+    const TagLib::ID3v2::FrameList albumArtistFrame(tag.frameListMap()["TPE2"]);
+    if (!albumArtistFrame.isEmpty()) {
+        pTrackMetadata->setAlbumArtist(toQStringFirst(albumArtistFrame));
+    }
+
+    if (pTrackMetadata->getAlbum().isEmpty()) {
+        const TagLib::ID3v2::FrameList originalAlbumFrame(
+                tag.frameListMap()["TOAL"]);
+        pTrackMetadata->setAlbum(toQStringFirst(originalAlbumFrame));
+    }
+
+    const TagLib::ID3v2::FrameList composerFrame(tag.frameListMap()["TCOM"]);
+    if (!composerFrame.isEmpty()) {
+        pTrackMetadata->setComposer(toQStringFirst(composerFrame));
+    }
+
+    const TagLib::ID3v2::FrameList groupingFrame(tag.frameListMap()["TIT1"]);
+    if (!groupingFrame.isEmpty()) {
+        pTrackMetadata->setGrouping(toQStringFirst(groupingFrame));
+    }
+
+    // ID3v2.4.0: TDRC replaces TYER + TDAT
+    const TagLib::ID3v2::FrameList recordingDateFrame(
+            tag.frameListMap()["TDRC"]);
+    if (!recordingDateFrame.isEmpty()) {
+        pTrackMetadata->setYear(toQStringFirst(recordingDateFrame));
+    }
+}
+
+void readAPETag(TrackMetadata* pTrackMetadata, const TagLib::APE::Tag& tag) {
+    if (kDebugMetadata) {
+        for (TagLib::APE::ItemListMap::ConstIterator it(
+                tag.itemListMap().begin()); it != tag.itemListMap().end();
+                ++it) {
+            qDebug() << "APE" << toQString((*it).first) << "-"
+                    << toQString((*it).second.toString());
+        }
+    }
+
+    readTag(pTrackMetadata, tag);
+
+    if (tag.itemListMap().contains("BPM")) {
+        pTrackMetadata->setBpmString(toQString(tag.itemListMap()["BPM"]));
+    }
+
+    if (tag.itemListMap().contains("REPLAYGAIN_ALBUM_GAIN")) {
+        pTrackMetadata->setReplayGainDbString(
+                toQString(tag.itemListMap()["REPLAYGAIN_ALBUM_GAIN"]));
+    }
+    //Prefer track gain over album gain.
+    if (tag.itemListMap().contains("REPLAYGAIN_TRACK_GAIN")) {
+        pTrackMetadata->setReplayGainDbString(
+                toQString(tag.itemListMap()["REPLAYGAIN_TRACK_GAIN"]));
+    }
+
+    if (tag.itemListMap().contains("Album Artist")) {
+        pTrackMetadata->setAlbumArtist(
+                toQString(tag.itemListMap()["Album Artist"]));
+    }
+
+    if (tag.itemListMap().contains("Composer")) {
+        pTrackMetadata->setComposer(toQString(tag.itemListMap()["Composer"]));
+    }
+
+    if (tag.itemListMap().contains("Grouping")) {
+        pTrackMetadata->setGrouping(toQString(tag.itemListMap()["Grouping"]));
+    }
+
+    if (tag.itemListMap().contains("Year")) {
```

Comment:
This is the upwards compatible namer for "Release date"  ISO 8601 formated 
http://wiki.hydrogenaud.io/index.php?title=APE_Item_Value

Id3 required the ISO 8601 with separator "T" 

Unfortunately that does not  match TDRC Recording time which we use in the id3 case above. 
This here matches TDRL Release time

However Musicbrainz stores "Release Date" to "TDRC Recording time" so it seams to be a common practice 
IMHO this is a good reference:  
https://picard.musicbrainz.org/docs/mappings/

Conclusion: 
Current solution works fine. We just need some comments in code for later reviews. 
// We do it like Musicbrainz ...